spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisQueueMessageDrivenEndpoint.java

```diff
@@ -158,6 +160,16 @@ public void setRecoveryInterval(long recoveryInterval) {
 		this.recoveryInterval = recoveryInterval;
 	}
 
+	/**
+	 * Should data from the Redis queue be read from the left end of the queue (using leftPop)?
+	 * This is useful when the queue is fed by a software that uses rightPush.
```

Comment:
regarding my comment above, this is an interesting twist as it is still FIFO in the case that a producer is using right-push... my concern is that the LIST type in Redis can either be treated as a Queue or a Stack depending on right/left-pop/push behavior. When using the Spring Integration redis `outbound-queue` AND `inbound-queue` adapters, the behavior is consistent, but IIUC this is really addressing a case where the producer is some other system that does not treat it as a traditional (left-to-right ordered) FIFO queue.